app/models/theme.rb

```diff
@@ -156,8 +156,10 @@ def self.transform_ids(ids, extend: true)
 
       all_ids = [parent, *components]
 
-      disabled_ids = Theme.where(id: all_ids).includes(:remote_theme)
-        .reject(&:enabled?).pluck(:id)
+      disabled_ids = Theme.where(id: all_ids)
+        .includes(:remote_theme)
+        .select { |t| !t.supported? || t.disabled? }
```

Comment:
This select is a bit confusing to me ... can't it use a `.where` here?

Reply:
`.supported?` isn't a column it's a method that checks if the theme version is compatible with discourse's, so we need to load up the themes and filter out the ones that aren't compatible or disabled. Does that make sense?

Reply:
yes, makes sense... what I got confused here about is the way it does a `.pluck` here afterwards. 

```
 pry(main)> Topic.limit(10).select{|t| t.created_at < 1.year.ago}.class
  Topic Load (1.4ms)  SELECT  "topics".* FROM "topics" WHERE ("topics"."deleted_at" IS NULL) LIMIT 10
=> Array

 [Topic.new].pluck(:id)
=> [nil]
```

I guess active support likes monkey patching.